src/platform/Darwin/DnssdContexts.cpp

```diff
@@ -602,15 +605,47 @@ bool ResolveContext::TryReportingResultsForInterfaceIndex(uint32_t interfaceInde
     return true;
 }
 
-CHIP_ERROR ResolveContext::OnNewAddress(uint32_t interfaceId, const struct sockaddr * address)
+bool ResolveContext::TryReportingResultsForInterfaceIndex(uint32_t interfaceIndex)
+{
+    if (interfaceIndex == 0)
+    {
+        // Not actually an interface we have.
+        return false;
+    }
+
+    for (auto & interface : interfaces)
+    {
+        if (interface.first.interfaceId == interfaceIndex)
+        {
+            auto & ips = interface.second.addresses;
```

Comment:
Again, why are we checking `ips` here?  Just call the other overload; it does this check already.

Reply:
done removed.